dlib/geometry/polygon.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2022  Davis E. King (davis@dlib.net), Adrià Arrufat
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_POLYGOn_
+#define DLIB_POLYGOn_
+
+#include "polygon_abstract.h"
+#include "rectangle.h"
+#include "vector.h"
+
+namespace dlib
+{
+    class polygon
+    {
+    public:
+        using size_type = std::vector<point>::size_type;
+
+        polygon(const std::vector<point>& points) : points(points) {}
```

Comment:
```suggestion
        polygon(std::vector<point> points) : points(std::move(points)) {}
```
That way it can move sometimes vs always forcing a copy.